.github/hs-deploy-action/src/functions.ts

```diff
@@ -0,0 +1,76 @@
+import * as child_process from 'child_process';
+import * as exec from '@actions/exec';
+
+export async function build() {
+  //build libs and app
+  await exec.exec('yarn', ['install']);
+  await exec.exec('yarn', ['build:libs']);
+  await exec.exec('yarn', ['build']);
+}
+
+export async function deploy(github: any, octoKit: any) {
+  const context = github.context;
+  const branch = context.payload.pull_request.head.ref;
+
+  console.log(`--> Deploying branch ${branch}`);
+  const prNumber = branch.substring('feature/GH-'.length, branch.length);
+
+  const bundleId = `spartacus-feature-spa${prNumber}`;
+  const command = `upp application deploy -b ${bundleId} -t spartacus -s ./dist/storefrontapp -e stage`;
+
+  const exp = /https\:\/\/\w+\.cloudfront\.net/;
+  let output = '';
+
+  const process = child_process.exec(command);
+
+  process.stdout?.on('data', (data) => {
+    const line = data.toString();
+    const match = line.match(exp);
+    if (match && match.length > 0) {
+      const body = `:rocket: Spartacus deployed to [${match}](${match})`;
+      addComment(context, octoKit, body);
```

Comment:
Can we instead of a comment use the github deployments feature? Seems better suited for this action.

Reply:
I could not find how to add a comment there. Do you know how?